src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstruct.cs

```diff
@@ -59,6 +61,27 @@ protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymb
 
         protected override int GetParameterCountFromSyntax() => _ctor.ParameterCount;
 
+        protected override void MethodChecks(BindingDiagnosticBag diagnostics)
+        {
+            base.MethodChecks(diagnostics);
+
+            if (ParameterCount == _properties.Length)
```

Comment:
- When can the this condition be false?


- What if the Deconstruct method is declared explicitly:

    ```csharp
    public record Base
    {
        public int I { get; set; } = 42;
        public Base(int ignored) { }
    
        public void Deconstruct(out string s, out string s2) { s = s2 = "";  }
    }
    public record C(int I) : Base(I)
    {
        public void I() { }
    
        public void Deconstruct(out string s, out string s2) { s = s2 = "";  }
    }
    ```

Reply:
- the condition is false when `addProperties` in `SourceMemberContainerSymbol` doesn't find a property for each parameter (we produce `ERR_BadRecordMemberForPositionalParameter` in that case).
- thanks. I need to think more about scenarios where `Deconstruct` is declared in source. We should produce a diagnostic too, but currently don't.

---
In reply to: [614614357](https://github.com/dotnet/roslyn/pull/52660#discussion_r614614357) [](ancestors = 614614357)